cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/SchemaService.java

```diff
@@ -235,20 +237,9 @@ private String getSchemaCompatibility(
       Pair<String, RestTemplate> reqDetails =
           clusterApiUtils.getRequestDetails(suffixUrl, protocol);
 
-      Map<String, String> params = new HashMap<>();
-
-      HttpHeaders headers =
-          clusterApiUtils.createHeaders(clusterIdentification, KafkaClustersType.SCHEMA_REGISTRY);
-      HttpEntity<Object> request = new HttpEntity<>(headers);
       ResponseEntity<Map<String, String>> responseList =
-          reqDetails
-              .getRight()
-              .exchange(
-                  reqDetails.getLeft(),
-                  HttpMethod.GET,
-                  request,
-                  GET_SCHEMA_COMPATIBILITY_TYPEREF,
-                  params);
+          getSubjectSchemaCompatibility(
+              reqDetails, new HashMap<>(), createSchmeaRegistryRequest(clusterIdentification));
       log.info("Schema compatibility " + responseList);
       return responseList.getBody().get("compatibilityLevel");
```

Comment:
Any possibility for this returning null ?

Reply:
There is a small chance that the schema registry would return a different object but I think that would be a breaking change, It would be caught at this level by the try catch however and gracefully handled. I dont think this one needs a change.